src/combinations.rs

```diff
@@ -135,18 +180,18 @@ impl<I: Iterator> Combinations<I> {
         }
 
         // Increment index, and reset the ones to its right
-        self.indices[i] += 1;
-        for j in i + 1..self.indices.len() {
-            self.indices[j] = self.indices[j - 1] + 1;
+        indices[i] += 1;
+        for j in i + 1..indices.len() {
+            indices[j] = indices[j - 1] + 1;
         }
-
         // If we've made it this far, we haven't run out of combos
         false
     }
 
     /// Returns the n-th item or the number of successful steps.
-    pub(crate) fn try_nth(&mut self, n: usize) -> Result<<Self as Iterator>::Item, usize>
+    pub(crate) fn try_nth(&mut self, n: usize) -> Result<Idx::Item, usize>
```

Comment:
I'm not sure if `Idx::Item` is really better than `<Self as Iterator>::Item` here. Sure, it's less work to type it, but I think `try_nth` is a method on the iterator thing, so its result does not need to include the (inner) `Idx` type.